miden-lib/src/auth.rs

```diff
@@ -0,0 +1,10 @@
+use miden_objects::crypto::dsa::rpo_falcon512;
+
+/// Defines authentication schemes available to standard and faucet accounts.
+/// At the moment only RpoFalcon512 exists.
+pub enum AuthScheme {
+    /// RPO Falcon512: a variant of the Falcon signature scheme. This variant differs
+    /// from the standard in that instead of using SHAKE256 hash function in the hash-to-point
+    /// algorithm we use RPO256. This makes the signature more efficient to verify in Miden VM.
+    RpoFalcon512 { pub_key: rpo_falcon512::PublicKey },
```

Comment:
I would probably remove "At the moment.." as it is pretty clear from the enum definition.

Also, I would write the comment for `RpoFalcon512` as follow:
```
/// A single-key authentication scheme which relies RPO Falcon512 signatures. RPO Falcon512 is a
/// variant of the [Falcon](https://falcon-sign.info/) signature scheme. This variant differs from
/// the standard in that instead of using SHAKE256 hash function in the hash-to-point algorithm we
/// use RPO256. This makes the signature more efficient to verify in Miden VM.
```